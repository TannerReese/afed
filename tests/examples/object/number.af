# Test rational and real arithmetic

# Multiline expression
4 + 0
+
5 - 1
+
-6
=
``,

# Rational arithmetic
48 / 32 + (100 / 70)^(6/3) = ``,
2 / 3 - 5/7 = ``,
23 // (4/5) = ``,
23 // (4 / 5) % (43 / 47) = ``,

# Real arithmetic
-  1 + 2 *- 3.1 = ``,
67 ^ (2 / 3) * 67^-1 - 3.1 = ``,
x: 2 ^ 1.1 ^ 1.01 ^ (23 % 2.1) = ``,
34 // 2.45 % x / 3 * 15 % -0.1 = ``,

# Equality
23 / 7 == 69 / 21 = ``,
34.0 / 56 != 0.607142 = ``,
3.14 == 3.13999999999 = ``,
0 == 1 / 100000 = ``,

# Ordering
-5 >= -5 = ``,
3.4 > 5.6 = ``,
2 / 4 <= 5 / 6 = ``,
2.3 - 4 ^ 2 < -2 * 5 ^ 1.2 = ``,

