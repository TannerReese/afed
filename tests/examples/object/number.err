line 37, column 10 of "number.af" Eval Error: Binary operator && not implemented between 2 and 3
line 38, column 13 of "number.af" Eval Error: Binary operator || not implemented between -7 and 2.3
line 39, column 6 of "number.af" Eval Error: Unary operator ! not implemented for type number
line 40, column 11 of "number.af" Eval Error: Cannot call type number
4 Eval Errors encountered
