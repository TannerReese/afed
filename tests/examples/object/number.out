# Test rational and real arithmetic

# Multiline expression
4 + 0
+
5 - 1
+
-6
=
`2`,

# Rational arithmetic
48 / 32 + (100 / 70)^(6/3) = `347 / 98`,
2 / 3 - 5/7 = `-1 / 21`,
23 // (4/5) = `28`,
23 // (4 / 5) % (43 / 47) = `26 / 47`,

# Real arithmetic
-  1 + 2 *- 3.1 = `-7.2`,
67 ^ (2 / 3) * 67^-1 - 3.1 = `-2.8537884631009867`,
x: 2 ^ 1.1 ^ 1.01 ^ (23 % 2.1) = `2.1466832429584723`,
34 // 2.45 % x / 3 * 15 % -0.1 = `0.09950271124582952`,

# Equality
23 / 7 == 69 / 21 = `true`,
34.0 / 56 != 0.607142 = `true`,
3.14 == 3.13999999999 = `true`,
0 == 1 / 100000 = `false`,

# Ordering
-5 >= -5 = `true`,
3.4 > 5.6 = `false`,
2 / 4 <= 5 / 6 = `true`,
2.3 - 4 ^ 2 < -2 * 5 ^ 1.2 = `false`,

# Number unimplemented operators
2 && 3 = `Eval Error: Binary operator && not implemented between 2 and 3`,
-7 || 2.3 = `Eval Error: Binary operator || not implemented between -7 and 2.3`,
!3 = `Eval Error: Unary operator ! not implemented for type number`,
4 $ "a" = `Eval Error: Cannot call type number`,

